src/Coordination/KeeperStorage.cpp

```diff
@@ -454,6 +454,14 @@ void KeeperMemNode::shallowCopy(const KeeperMemNode & other)
     cached_digest = other.cached_digest;
 }
 
+KeeperMemNode KeeperMemNode::copyFromSnapshotNode()
+{
+    KeeperMemNode node_copy;
+    node_copy.shallowCopy(*this);
+    node_copy.children = std::move(children);
```

Comment:
Why can we `move(children)`? Doesn't it make the current object malformed?

Reply:
In `shallowCopy` we copy `stats` but not `children`, so `getChildren` on on object with was obtained via `shallowCopy` will trigger assert, is it intended?

Reply:
It is a bit hacky, when creating snapshot we don't iterate children.
If snapshot is being made, each update of a node will create a copy first and then update it.
Snapshot will use the old node, while every new operation will use new node.
So that's why we can move children from the snapshotted node, we don't need it.

Reply:
As a safety I added asserts on children getters (`getChildren`)